cmd/autofix.go

```diff
@@ -0,0 +1,159 @@
+			results = append(results, result)
+		}
+	}
+	return results
+}
+
+func fixSecurityContextNIL(resource k8sRuntime.Object) k8sRuntime.Object {
+	var containers []Container
+	for _, container := range getContainers(resource) {
+		container.SecurityContext = &SecurityContext{
+			Capabilities: &Capabilities{
+				Drop: []Capability{"AUDIT_WRITE", "CHOWN", "DAC_OVERRIDE", "FOWNER",
+					"FSETID", "KILL", "MKNOD", "NET_BIND_SERVICE", "NET_RAW", "SETFCAP",
+					"SETGID", "SETUID", "SETPCAP", "SYS_CHROOT"},
+			},
+			Privileged: newFalse(),
+			//SELinuxOptions *SELinuxOptions
+			//RunAsUser *int64
+			RunAsNonRoot:             newTrue(),
+			ReadOnlyRootFilesystem:   newTrue(),
+			AllowPrivilegeEscalation: newFalse(),
+		}
+		containers = append(containers, container)
+	}
+	return setContainers(resource, containers)
+}
+
+func fixCapabilities(resource k8sRuntime.Object) k8sRuntime.Object {
+	var containers []Container
+	for _, container := range getContainers(resource) {
+		container.SecurityContext.Capabilities = &Capabilities{
+			Drop: []Capability{"AUDIT_WRITE", "CHOWN", "DAC_OVERRIDE", "FOWNER",
+				"FSETID", "KILL", "MKNOD", "NET_BIND_SERVICE", "NET_RAW", "SETFCAP",
+				"SETGID", "SETUID", "SETPCAP", "SYS_CHROOT"},
+		}
+		containers = append(containers, container)
+	}
+	return setContainers(resource, containers)
+}
+
+func fixPrivilegeEscalation(resource k8sRuntime.Object) k8sRuntime.Object {
+	var containers []Container
+	for _, container := range getContainers(resource) {
+		container.SecurityContext.Privileged = newFalse()
+		containers = append(containers, container)
+	}
+	return setContainers(resource, containers)
+}
+
+func fixAllowPrivilegeEscalation(resource k8sRuntime.Object) k8sRuntime.Object {
+	var containers []Container
+	for _, container := range getContainers(resource) {
+		container.SecurityContext.AllowPrivilegeEscalation = newFalse()
+		containers = append(containers, container)
+	}
+	return setContainers(resource, containers)
+}
+
+func fixReadOnlyRootFilesystem(resource k8sRuntime.Object) k8sRuntime.Object {
+	var containers []Container
+	for _, container := range getContainers(resource) {
+		container.SecurityContext.ReadOnlyRootFilesystem = newTrue()
+		containers = append(containers, container)
+	}
+	return setContainers(resource, containers)
+}
+
+func fixRunAsNonRoot(resource k8sRuntime.Object) k8sRuntime.Object {
+	var containers []Container
+	for _, container := range getContainers(resource) {
+		container.SecurityContext.ReadOnlyRootFilesystem = newTrue()
+		containers = append(containers, container)
+	}
+	return setContainers(resource, containers)
+}
+
+func fixServiceAccountToken(resource k8sRuntime.Object) k8sRuntime.Object {
+	return setASAT(resource, false)
+}
+
+func fixDeprecatedServiceAccount(resource k8sRuntime.Object) k8sRuntime.Object {
+	return disableDSA(resource)
+}
+
+func fixStuff(resource k8sRuntime.Object, result Result) (fixed k8sRuntime.Object) {
```

Comment:
A better name for this function like `fixPotentialSecurityIssues` or something similar

Reply:
@jinankjain It has the WIP prefix and label for a reason, there is a lot that needs to happen before this can be merged :) But I like the name you suggested.

Reply:
@klautcomputing Yeah I was just going through PR. It seems super interesting feature. 💯